Mathlib/Topology/Compactness/CompactlyCoherentSpace.lean

```diff
@@ -0,0 +1,336 @@
+-/
+
+import Mathlib.Topology.Defs.Induced
+import Mathlib.Topology.Coherent
+import Mathlib.Topology.Compactness.CompactlyGeneratedSpace
+
+/-!
+# Compactly coherent spaces and the k-ification
+
+In this file we will define compactly coherent spaces and the compact coherentification and prove
+basic properties about them. This is a weaker version of `CompactlyGeneratedSpace`.
+These notions agree on Hausdorff spaces. They are both referred to as compactly generated spaces in
+the literature while the compact coherentification is often called the compact coherentification.
+
+## Main definitions
+* `CompactlyCoherentSpace`: A compactly coherent space is a topological space in which a set `A` is
+  open iff for every compact set `B`, the intersection `A ∩ B` is open in `B`.
+* `CompactlyCoherentSpace.CompactCoherentification`: For a topological space `X` one can define
+  another topology on `X` as follows: `A` is open iff for all compact sets `B`, the intersection
+  `A ∩ B` is open in `B`.
+
+## Main results
+* `CompactlyCoherentSpace.of_weaklyLocallyCompactSpace`: every weakly locally compact space is a
+  compactly coherent space.
+* `CompactlyCoherentSpace.of_sequentialSpace`: every sequential space is a compactly coherent space.
+* `CompactlyCoherentSpace.of_compactlyGeneratedSpace`: every compactly generated space is a
+  compactly coherent space.
+* `CompactlyCoherentSpace.isCompact_iff_isCompact_in_compactCoherentification`: The compact sets of
+  a topological space and its compact coherentification agree.
+* `CompactlyCoherentSpace.CompactCoherentification.instCompactlyCoherentSpace`: The compact
+  coherentification makes any space into a compactly coherent space.
+* `CompactlyCoherentSpace.compactCoherentification_compactlyCoherentSpace_eq_self`: The compact
+  coherentification of a compactly coherent space `X` preserves the topology on `X`.
+* `CompactlyCoherentSpace.continuous_compactCoherentification_of_continuousOn_compact`: If a map
+  `f : X → Y` is continuous on every compact subset of `X` then it is continuous when viewed as a
+  map from `CompactCoherentification X` to `CompactCoherentification Y`.
+
+## References
+* [J. Munkres, *Topology*][Munkres2000]
+* <https://en.wikipedia.org/wiki/Compactly_generated_space>
+-/
+
+noncomputable section
+
+open Set Set.Notation Topology
+
+/-! ### Compactly coherent spaces-/
+
+/-- A space is a compactly coherent space if the topology is generated by the compact sets. -/
+class CompactlyCoherentSpace  (X : Type*) [TopologicalSpace X] : Prop where
+  /-- A space is a compactly coherent space if the topology is generated by the compact sets.-/
+  isCoherentWith : IsCoherentWith (X := X) {K | IsCompact K}
+
+namespace CompactlyCoherentSpace
+
+universe u
+
+variable {X : Type u} [TopologicalSpace X]
+
+/-- A set `A` in a compactly coherent space is open iff for every compact set `K`, the intersection
+`K ∩ A` is open in `K`. -/
+lemma isOpen_iff [CompactlyCoherentSpace X] (A : Set X) :
```

Comment:
I think `A` should be made implicit here, for consistency with the rest of the library.